cmd/booster-http/server.go

```diff
@@ -203,8 +203,14 @@ func (s *HttpServer) handleByPieceCid(w http.ResponseWriter, r *http.Request) {
 		// TODO: Allow this to be configurable
 		expectedPaymentAmount := big.NewInt(10)
 
-		if hasPaid := checkPaymentChannelBalance(s.nitroRpcClient, chId, expectedPaymentAmount); !hasPaid {
-			writeError(w, r, http.StatusPaymentRequired, "payment required")
+		hasPaid, err := checkPaymentChannelBalance(s.nitroRpcClient, chId, expectedPaymentAmount)
+		if err != nil {
+			writeError(w, r, http.StatusInternalServerError, err.Error())
+			return
+		}
+
+		if !hasPaid {
+			webError(w, fmt.Errorf("payment of %d required", expectedPaymentAmount.Uint64()), http.StatusPaymentRequired)
```

Comment:
What's the difference between `writeError` and `webError`?

Reply:
It looks like `webError` sets some additional headers related to CORS. I am changing this `webError` to `writeError` to conform to the rest of the error handling within this method.